x-pack/plugin/sql/qa/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/SysColumnsTestCase.java

```diff
@@ -267,6 +266,236 @@ public void testAliasWithSubfieldsAndDifferentRootFields() throws Exception {
+            .endObject();
+        });
+        
+        createIndexWithMapping("test2", builder -> {
+            builder.startObject("id").field("type", "keyword").endObject();
+            builder.startObject("name")
+                .field("type", "keyword")                               // <-------- first difference in mapping
+                .startObject("fields")
+                    .startObject("raw")
+                        .field("type", "keyword")
+                    .endObject()
+                .endObject()
+            .endObject();
+            builder.startObject("address")
+                .startObject("properties")
+                    .startObject("city")
+                        .field("type", "text")
+                        .startObject("fields")
+                            .startObject("raw")
+                                .field("type", "keyword")
+                            .endObject()
+                        .endObject()
+                    .endObject()
+                    .startObject("county")
+                        .field("type", "text")                          // <-------- second difference in mapping
+                        .startObject("fields")
+                            .startObject("raw")
+                                .field("type", "keyword")
+                            .endObject()
+                        .endObject()
+                    .endObject()
+               .endObject()
+            .endObject();
+        });
+
+        createAliases(builder -> {
+            builder.startObject().startObject("add").field("index", "test1").field("alias", "test_alias").endObject().endObject();
+            builder.startObject().startObject("add").field("index", "test2").field("alias", "test_alias").endObject().endObject(); 
+        });
+        
+        assertResultsForQuery("SYS COLUMNS", new String[][] {
+            {"test1"      ,"address.city"      ,"TEXT"},
+            {"test1"      ,"address.city.raw"  ,"KEYWORD"},
+            {"test1"      ,"address.county"    ,"KEYWORD"},
+            {"test1"      ,"address.county.raw","KEYWORD"},
+            {"test1"      ,"id"                ,"KEYWORD"},
+            {"test1"      ,"name"              ,"TEXT"},
+            {"test1"      ,"name.raw"          ,"KEYWORD"},
+            {"test2"      ,"address.city"      ,"TEXT"},
+            {"test2"      ,"address.city.raw"  ,"KEYWORD"},
+            {"test2"      ,"address.county"    ,"TEXT"},
+            {"test2"      ,"address.county.raw","KEYWORD"},
+            {"test2"      ,"id"                ,"KEYWORD"},
+            {"test2"      ,"name"              ,"KEYWORD"},
+            {"test2"      ,"name.raw"          ,"KEYWORD"},
+            {"test_alias" ,"address.city"      ,"TEXT"},
+            {"test_alias" ,"address.city.raw"  ,"KEYWORD"},
+            {"test_alias" ,"id"                ,"KEYWORD"}
+            // address.county gets removed since it has conflicting mappings
```

Comment:
That is incorrect - the field should show up as invalid. As should its subfields.
Otherwise test_alias looks like an invalid alias, potentially pointing to other indices.

Reply:
I'm not following. You mean we need to show in `SYS COLUMNS` all the fields (compatible or not)? In case of BI tools that use the output of SYS COLUMNS and build a SELECT X, Y, Z.... query will always result in an error if there will be incompatible fields in the SELECT.